src/router.js

```diff
@@ -327,6 +401,32 @@ var waitForChannelsAndIMs = function (callback) {
 };
 
 exports.serve = function (request, response) {
+
+  //If API request, there is a token in the request header which proves that 
+  //   the user is authenticated
+  if (request.headers.token){
+    username = request.headers.username;
```

Comment:
Is it possible not to send username in headers ? I think of something like this :
Store every token in a table inside a single spot of the NodeJS cache. Same in the browser cache.